todos/src/Components/TodosList.tsx

```diff
@@ -0,0 +1,34 @@
+import React from 'react'
+import Todo from './Todo'
+import {todo} from '../Types/todo'
+
+
+type TodoListProps = {
+    title: string
+    mode: string
+    todos: todo[]
+    submitCompleteTodo: (id: string) => void
+    submitInProgressTodo: (id: string) => void
+}
+
+const TodosList : React.FC <TodoListProps> = (props) => {
+
+    const pred = (todo : {id: string, active: boolean, label: string}) => {
+       return props.mode === "active" ? todo.active :
+        props.mode === "completed" ? !todo.active : true;
+    }
+
+    return (
+        <div>
+          <h2>{props.title}</h2>
+            {props.todos ? props.todos.map((todo) => 
+                pred(todo) ?  
```

Comment:
You need to filter things out, use filter and then map. Its important not to mix concerns

Reply:
It feels like the filtering should be done in the parent component.